src/model_signing/hashing.py

```diff
@@ -90,12 +90,17 @@ def __init__(self):
         symbolic link in the model directory results in an error.
         """
         self._ignored_paths = frozenset()
+        self._ignore_git_paths = True
         self.use_file_serialization()
 
     def hash(self, model_path: os.PathLike) -> manifest.Manifest:
         """Hashes a model using the current configuration."""
+        ignored_paths = [path for path in self._ignored_paths]
+        if self._ignore_git_paths:
+            ignored_paths.append(".git/")
```

Comment:
> We already check in `should_ignore` from `serialization.py` if a path is
relative to an ignored path, so that should already exclude the git
paths if `.git` is added to the excludes.

Isn't this slightly different though? As written, this ignores the top-level `.git` directory. Previously you ignored every `.git` and `.gitattributes`. I'm not saying that was better, just different

Reply:
Yes, but we actually want to only ignore the `.git` directory from the model root. It's very unlikely that there will be another `.git` directory somewhere deep inside the model.

Re `.gitattributes`, I'll add it.

Reply:
personally I think it was better just as `.git` and not the other files, I was mainly pointing out the change in behavior.

But happy to merge either way. going to give a few minutes for possible response then approve due to auto-merge.

Reply:
We always have the option to set this flag to false and add to exclude only a subset of these files. This is just a convenience method to ignore all at once.